quinn-proto/src/config/mod.rs

```diff
@@ -262,6 +276,37 @@ impl ServerConfig {
         self
     }
 
+    /// Duration after an address validation token was issued for which it's considered valid
+    ///
+    /// This refers only to tokens sent in NEW_TOKEN frames, in contrast to retry tokens.
+    ///
+    /// Defaults to 2 weeks.
+    pub fn validation_token_lifetime(&mut self, value: Duration) -> &mut Self {
+        self.validation_token_lifetime = value;
+        self
+    }
+
+    /// Set a custom [`TokenLog`]
+    ///
+    /// Setting this to `None` makes the server ignore all address validation tokens (that is,
+    /// tokens originating from NEW_TOKEN frames--retry tokens may still be accepted).
+    ///
+    /// Defaults to a `None`.
+    pub fn validation_token_log(&mut self, log: Option<Arc<dyn TokenLog>>) -> &mut Self {
+        self.validation_token_log = log;
+        self
+    }
+
+    /// Number of address validation tokens sent to a client when its path is validated
+    ///
+    /// This refers only to tokens sent in NEW_TOKEN frames, in contrast to retry tokens.
+    ///
+    /// Defaults to 2.
```

Comment:
I didn't put too much thought into the default for this commit, since it's immediately followed by a commit that adds a default token log. However, there's an argument to be made that "send validation tokens then ignore them when presented" is justified: It costs very little for the server to send the client a token or to receive and ignore a token. However, even if the server doesn't currently have a token log, the fact that able-to-utilize-new-token clients have been receiving tokens for the last 2 weeks means that those tokens could start becoming useful the moment the server gets reconfigured to have a token log, rather than having to wait for clients to start accumulating cached tokens after the server gains a token log. So, risk is negligible, and reward is conceivable.

Reply:
That's a reasonable argument. Can we include it in the documentation for the benefit of users/our future selves?

Reply:
I changed my mind. For the benefit to be realizable, one would have to configure persistent token encryption keys. If they are doing that, they could be reasonably expected to also change this setting too. Plus, there may be costs to clients in the form of cache eviction even if the server bears no costs.